### ISSUE TITLE
Add `where` argument to `nn.logsumexp`

### ISSUE THREAD

**Issue by carlosgmartin** [google/jax#20555]
The following functions in [`jax.nn`](https://jax.readthedocs.io/en/latest/jax.nn.html#other-functions) receive a `where` argument:
- [`softmax`](https://jax.readthedocs.io/en/latest/_autosummary/jax.nn.softmax.html)
- [`log_softmax`](https://jax.readthedocs.io/en/latest/_autosummary/jax.nn.log_softmax.html)
- [`standardize`](https://jax.readthedocs.io/en/latest/_autosummary/jax.nn.standardize.html)

**Feature request:** Add a `where` argument to the following functions:
- [ ] [`logsumexp`](https://jax.readthedocs.io/en/latest/_autosummary/jax.nn.logsumexp.html)

I can submit a PR for this.

**Comment by jakevdp** [comment]
Sounds good! Feel free to tag me for review if you put together a PR

**Comment by carlosgmartin** [comment]
@jakevdp Done: #20578.
